src/TLabel.cpp

```diff
@@ -88,23 +93,29 @@ void TLabel::mousePressEvent(QMouseEvent* event)
     if (forwardEventToMapper(event)) {
         return;
     }
-    if (mMouseButtons.contains(event->button())) {
-        if (mpHost) {
-            TEvent tmpClickParams = mClickParams;
-            tmpClickParams.mArgumentList.append(mMouseButtons.value(event->button()));
-            tmpClickParams.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
-            mpHost->getLuaInterpreter()->callEventHandler(mClick, tmpClickParams);
-        }
-        event->accept();
-        return;
+
+    if (mpHost) {
+        mpHost->getLuaInterpreter()->callEventHandler(mClick, mClickParams, event);
     }
+    event->accept();
+    return;
+
 
     QWidget::mousePressEvent(event);
```

Comment:
I'm not quite sure, but if I read the function correctly, this line is unreachable.